packages/storage/amplify_storage_s3/example/lib/main.dart

```diff
@@ -105,19 +108,249 @@ class HomeScreen extends StatefulWidget {
 }
 
 class _HomeScreenState extends State<HomeScreen> {
+  List<StorageItem> list = [];
+  var imageUrl = '';
+
+  @override
+  void initState() {
+    super.initState();
+    _checkAuthStatus();
+    _listAllPublicFiles();
+  }
+
+  // sign out of the app
+  Future<void> _signOut() async {
+    try {
+      await Amplify.Auth.signOut();
+      _logger.debug('Signed out');
+    } on AuthException catch (e) {
+      _logger.error('Could not sign out - ${e.message}');
+    }
+  }
+
+  // check if the user is signed in
+  Future<void> _checkAuthStatus() async {
+    try {
+      final session = await Amplify.Auth.fetchAuthSession();
+      _logger.debug('Signed in: ${session.isSignedIn}');
+    } on AuthException catch (e) {
+      _logger.error('Could not check auth status - ${e.message}');
+    }
+  }
+
+  // upload a file to the S3 bucket
+  Future<void> _uploadFile() async {
+    final result = await FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: ['jpg', 'png'],
+      withReadStream: true,
+      withData: false,
+    );
+
+    if (result == null) {
+      _logger.debug('No file selected');
+      return;
+    }
+
+    final platformFile = result.files.single;
+
+    try {
+      await Amplify.Storage.uploadFile(
+        localFile: AWSFile.fromStream(
+          platformFile.readStream!,
+          size: platformFile.size,
+        ),
+        key: platformFile.name,
+        onProgress: (p) =>
+            _logger.debug('Uploading: ${p.transferredBytes}/${p.totalBytes}'),
+      ).result;
+      await _listAllPublicFiles();
+    } on StorageException catch (e) {
+      _logger.error('Error uploading file - ${e.message}');
+    }
+  }
+
+  // list all files in the S3 bucket
+  Future<void> _listAllPublicFiles() async {
+    try {
+      final result = await Amplify.Storage.list(
+        options: const StorageListOptions(
+          accessLevel: StorageAccessLevel.guest,
+          pluginOptions: S3ListPluginOptions.listAll(),
+        ),
+      ).result;
+      setState(() {
+        list = result.items;
+      });
+    } on StorageException catch (e) {
+      _logger.error('List error - ${e.message}');
+    }
+  }
+
+  // download file on mobile
+  Future<void> downloadFileMobile(String key) async {
+    final documentsDir = await getApplicationDocumentsDirectory();
+    final filepath = '${documentsDir.path}/$key';
+    try {
+      await Amplify.Storage.downloadFile(
+        key: key,
+        localFile: AWSFile.fromPath(filepath),
+        onProgress: (p0) => _logger
+            .debug('Progress: ${p0.transferredBytes} / ${p0.totalBytes}%)'),
```

Comment:
The percentage sign doesn't make sense here. The trailing `)` is also not needed

```console
I/flutter (17111): DEBUG | MyStorageApp | Progress: 2509391 / 2531037%)
I/flutter (17111): DEBUG | MyStorageApp | Progress: 2517582 / 2531037%)
I/flutter (17111): DEBUG | MyStorageApp | Progress: 2517584 / 2531037%)
I/flutter (17111): DEBUG | MyStorageApp | Progress: 2525775 / 2531037%)
```